pkg/controllers/disruption/singlenodeconsolidation_test.go

```diff
@@ -0,0 +1,303 @@
+var nodePoolInstanceTypeMap map[string]map[string]*cloudprovider.InstanceType
+
+var _ = Describe("SingleNodeConsolidation", func() {
+	BeforeEach(func() {
+		nodePool1 = test.NodePool(v1.NodePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "nodepool-1",
+			},
+			Spec: v1.NodePoolSpec{
+				Disruption: v1.Disruption{
+					ConsolidationPolicy: v1.ConsolidationPolicyWhenEmptyOrUnderutilized,
+					ConsolidateAfter:    v1.MustParseNillableDuration("0s"),
+				},
+			},
+		})
+		nodePool2 = test.NodePool(v1.NodePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "nodepool-2",
+			},
+			Spec: v1.NodePoolSpec{
+				Disruption: v1.Disruption{
+					ConsolidationPolicy: v1.ConsolidationPolicyWhenEmptyOrUnderutilized,
+					ConsolidateAfter:    v1.MustParseNillableDuration("0s"),
+				},
+			},
+		})
+		nodePool3 = test.NodePool(v1.NodePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "nodepool-3",
+			},
+			Spec: v1.NodePoolSpec{
+				Disruption: v1.Disruption{
+					ConsolidationPolicy: v1.ConsolidationPolicyWhenEmptyOrUnderutilized,
+					ConsolidateAfter:    v1.MustParseNillableDuration("0s"),
+				},
+			},
+		})
+		ExpectApplied(ctx, env.Client, nodePool1, nodePool2, nodePool3)
+
+		// Set up NodePool maps for candidate creation
+		nodePoolMap = map[string]*v1.NodePool{
+			nodePool1.Name: nodePool1,
+			nodePool2.Name: nodePool2,
+			nodePool3.Name: nodePool3,
+		}
+		nodePoolInstanceTypeMap = map[string]map[string]*cloudprovider.InstanceType{
+			nodePool1.Name: {leastExpensiveInstance.Name: leastExpensiveInstance},
+			nodePool2.Name: {leastExpensiveInstance.Name: leastExpensiveInstance},
+			nodePool3.Name: {leastExpensiveInstance.Name: leastExpensiveInstance},
+		}
+
+		// Create a single node consolidation controller
+		baseConsolidation := disruption.MakeConsolidation(fakeClock, cluster, env.Client, prov, cloudProvider, recorder, queue)
+		consolidation = disruption.NewSingleNodeConsolidation(baseConsolidation)
+	})
+
+	AfterEach(func() {
+		disruption.SingleNodeConsolidationTimeoutDuration = 3 * time.Minute
+		fakeClock.SetTime(time.Now())
+		ExpectCleanedUp(ctx, env.Client)
+	})
+
+	Context("candidate shuffling", func() {
+		It("should sort candidates by disruption cost", func() {
+			candidates, err := createCandidates(1.0, 3)
+			Expect(err).To(BeNil())
+
+			sortedCandidates := consolidation.SortCandidates(ctx, candidates)
+
+			// Verify candidates are sorted by disruption cost
+			Expect(sortedCandidates).To(HaveLen(9))
+			for i := 0; i < len(sortedCandidates)-1; i++ {
+				Expect(sortedCandidates[i].DisruptionCost).To(BeNumerically("<=", sortedCandidates[i+1].DisruptionCost))
+			}
+		})
+
+		It("should prioritize nodepools that timed out in previous runs", func() {
+			candidates, err := createCandidates(1.0, 3)
+			Expect(err).To(BeNil())
+
+			consolidation.PreviouslyUnseenNodePools.Insert(nodePool2.Name)
+			fmt.Println("consolidation.PreviouslyUnseenNodePools", consolidation.PreviouslyUnseenNodePools)
```

Comment:
```suggestion
			fmt.Println("consolidation.PreviouslyUnseenNodePools", consolidation.PreviouslyUnseenNodePools)
```

nit: Should probably drop this line now